### PR TITLE
swarm/api: correct bzz.resolve functionality

### PR DESCRIPTION
**Pull request by AgeManning**
This addresses issue #3477 

Tried to make it consistent with the current structure. I moved the logic into a separate file (ens.go), as it didn't seem to logically fit inside api.go. The resolve function is based off Api.Resolve as the swam docs indicate that bzz.resolve() should only resolve domains and not hashes.

### CODE REVIEW

File: swarm/api/ens.go [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+	"github.com/ethereum/go-ethereum/logger/glog"
+	"github.com/ethereum/go-ethereum/logger"
+)
+
+// Implements an RPC service
+type Ens struct {
+	api *Api
+}
+
+func NewEns(api *Api) *Ens {
```
Comment (zelig) [inline swarm/api/ens.go]: you do not need the api here.
In fact you do not need a new Ens struct either. 
Just use api.dns directly

**Comment by zelig** [comment]
@AgeManning please amend and check if it works. Do you know why it is not working now?

**Comment by AgeManning** [comment]
@zelig - I had considered doing this initially, the reasons I made a new struct were: 

1.  If we use api.dns directly the RPC server registers all methods and subscriptions associated with that object under the bzz namespace. In this instance, this includes the functions:
SetOwner
SetResolver
SetSubnodeOwner
Resiter
Resolver
SetContentHash

With the other objects exposed to the bzz namespace (i.e Info, Storage), there are associated functions defined in internal/web3ext/web3ext.go. 
I wasn't sure if it was correct to just give the RPC server api.dns and let it register all these unused functions, hence I made a new object with just a single function, Resolve. 

2. The logic seemed to be to build a structure, with the required functions used in the bzz namespace, as in Storage.go. Following this, I made ens.go, even though it had a single function, if you wanted more functions later related to ens, eg register() the logic would be separated. 

Anyway, the changes are much simpler, with the exception that more functions are registered under bzz. These changes seem to work for me.  :)

**Comment by fjl** [comment]
This is too old to be mergeable, and ENS integration has changed significantly. Please rebase and resubmit if you still need this change.

**Pull request closed without merging**